### PR TITLE
Add tests for Histogram metrics

### PR DESCRIPTION
**Pull request by cijothomas**
A small part to help kick start https://github.com/open-telemetry/opentelemetry-rust/issues/1817 
Also, stress test for Histograms. (Histograms have to update multiple memory points, so it must leverage different synchronization mechanisms, and hence must be tested separately)

### CODE REVIEW

File: stress/Cargo.toml [hunk 1]
```diff
@@ -4,11 +4,16 @@ version = "0.1.0"
 edition = "2021"
 publish = false
 
-[[bin]] # Bin to run the metrics stress tests
+[[bin]] # Bin to run the metrics stress tests for Counter
 name = "metrics"
```
Comment (TommyCpp) [inline stress/Cargo.toml:8]: Can we just rename this with _counter suffix?

Comment (cijothomas) [reply stress/Cargo.toml:8]: I renamed the files, and temporarily retaining "metrics" as bin name for counters, just to make it easy to run it, given it is run most frequently. once we are done with refactoring, will use full names

File: opentelemetry-sdk/src/metrics/mod.rs [hunk 2]
```diff
@@ -199,6 +200,20 @@ mod tests {
         counter_aggregation_helper(Temporality::Delta);
     }
 
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn histogram_aggregation_cumulative() {
+        // Run this test with stdout enabled to see output.
+        // cargo test histogram_aggregation_cumulative --features=metrics,testing -- --nocapture
```
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/mod.rs:206]: nit - this comment is valid for all the tests in general, do we need to add specifically for these tests ?

File: opentelemetry-sdk/src/metrics/mod.rs [hunk 3]
```diff
@@ -1007,6 +1022,65 @@ mod tests {
...
+        assert_eq!(data_point1.min.unwrap(), *values_kv1.iter().min().unwrap());
+        assert_eq!(data_point1.max.unwrap(), *values_kv1.iter().max().unwrap());
+
+        let data_point2 =
+            find_histogram_datapoint_with_key_value(&histogram.data_points, "key1", "value2")
+                .expect("datapoint with key1=value2 expected");
+        assert_eq!(data_point2.count, values_kv2.len() as u64);
+        assert_eq!(data_point2.sum, values_kv2.iter().sum::<u64>());
+        assert_eq!(data_point2.min.unwrap(), *values_kv2.iter().min().unwrap());
+        assert_eq!(data_point2.max.unwrap(), *values_kv2.iter().max().unwrap());
```
Comment (lalitb) [inline opentelemetry-sdk/src/metrics/mod.rs:1081]: I assume there would be separate tests to validate the delta/cumulative aggregation of these time-series data ?

**Review by lalitb — APPROVED** [review]
LGTM. Thanks.

File: opentelemetry-sdk/src/metrics/mod.rs (hunk 2 above)
Comment (cijothomas) [reply opentelemetry-sdk/src/metrics/mod.rs:206]: No strong opinion. I think it makes it easy to copy paste locally, to run one test alone. (I personally use it heavily)

File: opentelemetry-sdk/src/metrics/mod.rs (hunk 3 above)
Comment (cijothomas) [reply opentelemetry-sdk/src/metrics/mod.rs:1081]: absolutely! These basic tests check if the datapoint says it is Cumulative or Delta. Actual validation of the values - not yet done.

**Pull request merged**